sources/shimpei/src/main.rs

```diff
@@ -0,0 +1,87 @@
+/*!
+  shimpei is an OCI compatible shim wrapper around `oci-add-hooks`. Its sole purpose is
+  to call `oci-add-hooks` with the additional `--hook-config-path` and `--runtime-path`
+  parameters that can't be provided by containerd.
+*/
+
+#![deny(rust_2018_idioms)]
+
+#[macro_use]
+extern crate log;
+
+use nix;
+use simplelog::{Config as LogConfig, LevelFilter, SimpleLogger};
+use snafu::ResultExt;
+use std::env;
+use std::ffi::CString;
+use std::process;
+
+/// Path to runc binary
+const RUNC_BIN_PATH: &str = "/usr/bin/runc";
+
+/// Path to hooks definition, provided by runtime "vendor"
+const HOOK_CONFIG_PATH: &str = "/usr/share/oci-add-hooks/hook.json";
```

Comment:
Since this is actually vendor-specific, it might be worthwhile to hard-code the (only) vendor path we have today: `/usr/share/oci-add-hooks/nvidia.json`.  When we get to the point of adding a second vendor, we'd need a second compiled version of shimpei pointing to that second path anyway.

Reply:
I could make this change, but I would prefer if we don't hard-code the vendor specific name in shimpei. I understand that the contents of the `hook.json` file are vendor specific, but there won't be any case in which we have two "vendor runtimes" enabled in the same AMI, and shimpei doesn't really care about the hook that will be executed, it only knows that there is a `hook.json` file that `oci-add-hooks` will use.

The `hook.json` file can be provided by a `kmod-<kernel>-<vendor>` package, just like I did with `kmod-5.10-nvidia`, and we will save ourselves the N shimpei compilations, per vendor.

Reply:
We could avoid a second copy of `shimpei` by doing the multicall / busybox thing where we inspect arg0 and see if we're called as `nvidia-oci-hook`, `amdgpupro-hook`, etc, and use the matching `/usr/share/oci-add-hooks/<prefix>.json` as our input.

Then packages that wanted to use `shimpei` could just install a symlink, e.g. `/usr/bin/nvidia-oci-hook -> /usr/bin/shimpei`.